arch/cpu/native/net/tun6-net.c

```diff
@@ -95,35 +104,144 @@ ssystem(const char *fmt, ...)
   return system(cmd);
 }
 
+/*---------------------------------------------------------------------------*/
+static bool
+set_default_prefix(const char *prefix)
+{
+  char *ipaddr = strdup(prefix);
+  if(!ipaddr) {
+    return false;
+  }
+
+  uip_ip6addr_t prefix_addr;
+  bool success = false;
+  char *s = strchr(ipaddr, '/');
+  if(s) {
+    *s = '\0';
+    if(uiplib_ipaddrconv(ipaddr, &prefix_addr)) {
+      uip_ds6_set_default_prefix(&prefix_addr);
+      success = true;
+    }
+  }
+  free(ipaddr);
+  return success;
+}
+/*---------------------------------------------------------------------------*/
+const char *
+tun6_net_get_prefix(void)
+{
+  return config_ipaddr;
+}
+/*---------------------------------------------------------------------------*/
+void
+tun6_net_set_prefix(const char *prefix)
+{
+  if(!set_default_prefix(prefix)) {
+    LOG_WARN("Failed to set default prefix %s\n", prefix);
+  }
+  config_ipaddr = prefix;
```

Comment:
The variable is called prefix, should this assignment happen if the default prefix cannot be set?

Reply:
The native rpl-border-router should still work as long as the external network commands accept the configuration, and I tried to keep the behaviour close to the earlier behaviour. We could exit the application with an error instead. What do you think?

Reply:
If this is the old behavior, I suggest we keep it, too close to the release.

Reply:
Trying to set the default prefix at all is a new behaviour. There were no arguments to the native platform to set the prefix and tun device name earlier, but this allows multiple native applications with a network to run at the same time with their own sub-net. I leave it as is for now.